src/JMS/Serializer/Metadata/Driver/AnnotationDriver.php

```diff
@@ -240,12 +251,19 @@ public function loadMetadataForClass(\ReflectionClass $class)
                 if ((ExclusionPolicy::NONE === $exclusionPolicy && !$isExclude)
                     || (ExclusionPolicy::ALL === $exclusionPolicy && $isExpose)
                 ) {
-                    $propertyMetadata->setAccessor($accessType, $accessor[0], $accessor[1]);
+                    $propertyMetadata->setAccessor(
+                        $accessType,
+                        $accessor[0],
+                        $accessor[1],
+                        $accessTypeNaming
+                    );
                     $classMetadata->addPropertyMetadata($propertyMetadata);
                 }
             }
         }
 
+        $this->classUpdater->update($classMetadata);
```

Comment:
I'm not really sure that `classUpdater`, `ClassMetadataUpdater` and `update` method really reflects current responsibilities